diff --git a/make/Hsdis.gmk b/make/Hsdis.gmk
index 76695fc8dde..088c5f8f4b4 100644
--- a/make/Hsdis.gmk
+++ b/make/Hsdis.gmk
@@ -47,6 +47,9 @@ ifeq ($(HSDIS_BACKEND), capstone)
   else ifeq ($(call isTargetCpuArch, arm), true)
     CAPSTONE_ARCH := CS_ARCH_ARM
     CAPSTONE_MODE := CS_MODE_ARM
+  else ifeq ($(call isTargetCpuArch, riscv), true)
+    CAPSTONE_ARCH := CS_ARCH_RISCV
+    CAPSTONE_MODE := 'CS_MODE_RISCV64|CS_MODE_RISCVC'
   else
     $(error No support for Capstone on this platform)
   endif
